app/views/application/_nav_right.html.slim

```diff
@@ -1,27 +1,27 @@
 ul.nav.navbar-nav.navbar-right
-  li class=active?(:activities) = link_to 'Activities', root_path
-  li.dropdown class=dropdown_active?
+  li class=active_if(root_path) = link_to 'Activities', root_path
+  li.dropdown class=active_if_soc_dropdown_active
     a.dropdown-toggle data-toggle="dropdown" href="#"
       | Summer of Code
       span.caret
     ul.dropdown-menu role="menu"
       - if Season.projects_proposable?
         li.divider.hidden-xs
-        li class=active?(:projects) = link_to 'Submit your Project', new_project_path
-        li class=active?(:projects) = link_to 'Projects', projects_path      
-      - if show_application_link?
+        li class=active_if(new_project_path) = link_to 'Submit your Project', new_project_path
+        li class=active_if(projects_path) = link_to 'Projects', projects_path
+      - elsif during_application_phase?
         li.divider.hidden-xs
-        li class=active?(:application_drafts) = application_disambiguation_link
-        li class=active?(:projects) = link_to 'Projects', projects_path
-      - if current_season.started?
+        li class=active_if_controller('application_drafts') = application_disambiguation_link
```

Comment:
Is there a reason why you can't use the application_drafts_path here?
Because this one seems to be the only one that uses the `active_if_controller` helper.  Maybe we can do without that, or...?

Reply:
The `application_disambiguation_link` helper can return two different paths, depending on whether the user already has an `application_draft` or not. That's why a matching on the exact path is not possible 😞 
Both paths originate from the same controller though, that's why the controller approach still works here.

Reply:
Brilliant.